index/segment/mem/segment.go

```diff
@@ -369,3 +374,54 @@ func (s *segment) Close() error {
 	s.state.closed = true
 	return nil
 }
+
+func (s *segment) Seal() (sgmt.Segment, error) {
+	s.state.Lock()
+	defer s.state.Unlock()
+	if s.state.closed {
```

Comment:
nit: maybe it would be worth moving these checks into `checkIsOpenWithLock` `checkIsSealedWithLock` functions?

Reply:
sure

Reply:
actually, I think this hurts readability more than helping it - 
![screen shot 2018-05-12 at 11 55 53](https://user-images.githubusercontent.com/199982/39959146-80731180-55db-11e8-9692-bda3ee74377c.png)

Going to leave as is.

Reply:
Ahh, so what I meant was not have a check for sealed or closed but to coalesce them, i.e.

```
func (s * segment) checkIsOpenWithLock() bool {
  return !s.state.closed && ! s.state.sealed
}
```

Reply:
meh i can if you feel strongly but i think having the separate errors returned is also kinda nice.

Reply:
don't feel strongly, but you could also keep the separate errors:

```
func (s * segment) checkIsOpenWithLock() error {
  if s.state.closed {
    return errClosed
  }
  if s.state.sealed {
    return errSealed
  }
  return nil
}
```

Reply:
Sure